src/targets/gpu/mlir.cpp

```diff
@@ -636,7 +628,11 @@ struct mlir_program
         return op;
     }
 
-    void find_target() { target_arch = get_device_name(); }
+    void get_gpu_properties()
```

Comment:
This should be `set_gpu_properties`(as 'get' implies its returning a value). This also needs to take the `context` as a parameter.

Reply:
... then why didn't it take a context before?

But yeah, `set_target_properties()` is a better name here, agreed.